src/zenml/zen_stores/schemas/run_metadata_schemas.py

```diff
@@ -0,0 +1,155 @@
+from zenml.zen_stores.schemas.step_run_schemas import StepRunSchema
+from zenml.zen_stores.schemas.user_schemas import UserSchema
+
+
+class RunMetadataSchema(BaseSchema, table=True):
+    """SQL Model for run metadata."""
+
+    __tablename__ = "run_metadata"
+
+    pipeline_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=PipelineRunSchema.__tablename__,
+        source_column="pipeline_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    pipeline_run: "PipelineRunSchema" = Relationship(
+        back_populates="run_metadata"
+    )
+
+    step_run_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StepRunSchema.__tablename__,
+        source_column="step_run_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    step_run: "StepRunSchema" = Relationship(back_populates="run_metadata")
+
+    artifact_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=ArtifactSchema.__tablename__,
+        source_column="artifact_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=True,
+    )
+    artifact: "ArtifactSchema" = Relationship(back_populates="run_metadata")
+
+    stack_component_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=StackComponentSchema.__tablename__,
+        source_column="stack_component_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+    stack_component: "StackComponentSchema" = Relationship(
+        back_populates="run_metadata"
+    )
+
+    user_id: Optional[UUID] = build_foreign_key_field(
+        source=__tablename__,
+        target=UserSchema.__tablename__,
+        source_column="user_id",
+        target_column="id",
+        ondelete="SET NULL",
+        nullable=True,
+    )
+    user: "UserSchema" = Relationship(back_populates="run_metadata")
+
+    project_id: UUID = build_foreign_key_field(
+        source=__tablename__,
+        target=ProjectSchema.__tablename__,
+        source_column="project_id",
+        target_column="id",
+        ondelete="CASCADE",
+        nullable=False,
+    )
+    project: "ProjectSchema" = Relationship(back_populates="run_metadata")
+
+    key: str
+    value: str = Field(sa_column=Column(TEXT, nullable=False))
+    type: MetadataTypeEnum
+
+    def to_model(self) -> "RunMetadataResponseModel":
+        """Convert a `RunMetadataSchema` to a `RunMetadataResponseModel`.
+
+        Returns:
+            The created `RunMetadataResponseModel`.
+        """
+        return RunMetadataResponseModel(
+            id=self.id,
+            pipeline_run_id=self.pipeline_run_id,
+            step_run_id=self.step_run_id,
+            artifact_id=self.artifact_id,
+            stack_component_id=self.stack_component_id,
```

Comment:
```suggestion
            stack_component_id=self.stack_component_id if self.stack_component_id else None,
```

Stack components can be deleted

Reply:
Another good catch 👍

Reply:
Isn't it? 

    stack_component_id: Optional[UUID] = build_foreign_key_field(
        source=__tablename__,
        target=StackComponentSchema.__tablename__,
        source_column="stack_component_id",
        target_column="id",
        ondelete="SET NULL",
        nullable=True,
    )

Reply:
sorry, wrong wording, it is a reference, but not a `Relationship`.

I.e., if `self.stack_component_id` is `None`, then `self.stack_component_id if self.stack_component_id else None` will return `None`, which is the same as `self.stack_component_id` would give you anyways.